pym/bob/input.py

```diff
@@ -2015,10 +2015,12 @@ def prepare(self, pathFormatter, inputEnv, sandboxEnabled, inputStates, inputSan
         # filter indirect packages and add to result list
         tmp = indirectPackages
         indirectPackages = []
+        cache = {}
```

Comment:
By tracking it separately a package could now be in the direct *and* indirect dependency list of a package. That should not happen. If the package is in the dependency list then it is "direct". Otherwise it is "indirect".